src/components/ConversationSettings/ConversationAvatarEditor.vue

```diff
@@ -0,0 +1,310 @@
+				</NcButton>
+			</div>
+		</div>
+	</section>
+</template>
+
+<script>
+import VueCropper from 'vue-cropperjs'
+
+import axios from '@nextcloud/axios'
+import { getFilePickerBuilder, showError } from '@nextcloud/dialogs'
+import { generateUrl } from '@nextcloud/router'
+
+// import NcAvatar from '@nextcloud/vue/dist/Components/NcAvatar.js'
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+
+import ConversationIcon from '../ConversationIcon.vue'
+
+// eslint-disable-next-line n/no-extraneous-import
+import 'cropperjs/dist/cropper.css'
+
+import Upload from 'vue-material-design-icons/Upload.vue'
+import Folder from 'vue-material-design-icons/Folder.vue'
+import Delete from 'vue-material-design-icons/Delete.vue'
+
+const VALID_MIME_TYPES = ['image/png', 'image/jpeg']
+
+const picker = getFilePickerBuilder(t('spreed', 'Choose your profile picture'))
+	.setMultiSelect(false)
+	.setMimeTypeFilter(VALID_MIME_TYPES)
+	.setModal(true)
+	.setType(1)
+	.allowDirectories(false)
+	.build()
+
+export default {
+	name: 'ConversationAvatarEditor',
+
+	components: {
+		Delete,
+		Folder,
+		// NcAvatar,
+		NcButton,
+		Upload,
+		VueCropper,
+		ConversationIcon,
+	},
+
+	props: {
+		conversation: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			// Todo: get real value here
+			showCropper: false,
+			loading: false,
+			validMimeTypes: VALID_MIME_TYPES,
+			cropperOptions: {
+				aspectRatio: 1 / 1,
+				viewMode: 1,
+				guides: false,
+				center: false,
+				highlight: false,
+				autoCropArea: 1,
+				minContainerWidth: 300,
+				minContainerHeight: 300,
+			},
+		}
+	},
+
+	computed: {
+		inputId() {
+			return `account-property-${this.conversation.displayName}`
+		},
+
+		hasAvatar() {
+			return !!this.conversation.avatarVersion
+		},
+
+		fileTypeWarning() {
+			return t('spreed', 'The file must be a PNG or JPG')
+		},
+	},
+
+	methods: {
+		activateLocalFilePicker() {
+			// Set to null so that selecting the same file will trigger the change event
+			this.$refs.input.value = null
+			this.$refs.input.click()
+		},
+
+		onChange(e) {
+			this.loading = true
+			const file = e.target.files[0]
+			if (!this.validMimeTypes.includes(file.type)) {
+				showError(t('spreed', 'Please select a valid png or jpg file'))
+				this.cancel()
+				return
+			}
+
+			const reader = new FileReader()
+			reader.onload = (e) => {
+				this.$refs.cropper.replace(e.target.result)
+				this.showCropper = true
+			}
+			reader.readAsDataURL(file)
+		},
+
+		async openFilePicker() {
+			const path = await picker.pick()
+			this.loading = true
+			try {
+				const { data } = await axios.post(generateUrl('/avatar'), { path })
+				if (data.status === 'success') {
+					this.loading = false
+				} else if (data.data === 'notsquare') {
+					const tempAvatar = generateUrl('/avatar/tmp') + '?requesttoken=' + encodeURIComponent(OC.requestToken) + '#' + Math.floor(Math.random() * 1000)
```

Comment:
I may be wrong, but isn't `OC.requestToken` deprecated in favor of `getRequestToken` from [@nextcloud/auth](https://github.com/nextcloud/nextcloud-auth/)?